src/client-scopes/add/components/components.ts

```diff
@@ -0,0 +1,27 @@
+import type { FC } from "react";
+
+import type { ConfigPropertyRepresentation } from "@keycloak/keycloak-admin-client/lib/defs/authenticatorConfigInfoRepresentation";
+import { StringComponent } from "./StringComponent";
+import { BooleanComponent } from "./BooleanComponent";
+import { ListComponent } from "./ListComponent";
+import { RoleComponent } from "./RoleComponent";
+import { ScriptComponent } from "./ScriptComponent";
+
+export type ComponentProps = Omit<ConfigPropertyRepresentation, "type">;
+export const ComponentTypes: ReadonlyArray<string> = [
+  "String",
+  "boolean",
+  "List",
+  "Role",
+  "Script",
+];
```

Comment:
You can give this one a constant type so that it can be narrowed down further.
```suggestion
export const ComponentTypes = [
  "String",
  "boolean",
  "List",
  "Role",
  "Script",
] as const;
```

Reply:
I tried that first, but that doesn't work when you want to do:
```typescript
const componentType = property.type!;
if (ComponentTypes.includes(componentType)) {
  // only supported compnent types
```
because in this case when you use the `as const` then `compnentType` needs to be one of the types so defies the check

Reply:
Correct, I have a utility function in [another comment](https://github.com/keycloak/keycloak-admin-ui/pull/1187#discussion_r711106658) that fixes that.